internal/cmd/gocdk/init_test.go

```diff
@@ -147,6 +148,83 @@ func TestInit(t *testing.T) {
 		}
 	})
 }
+func TestInferModulePath(t *testing.T) {
+	t.Run("NoGOPATHEntry", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", testTempDirPrefix)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Error(err)
+			}
+		}()
+		ctx := context.Background()
+		pctx := &processContext{
+			workdir: dir,
+			env:     []string{},
+			stdin:   strings.NewReader(""),
+			stdout:  ioutil.Discard,
+			stderr:  ioutil.Discard,
+		}
+
+		fmt.Println(pctx.env)
+
+		const projectName = "myspecialproject"
+		if err := run(ctx, pctx, []string{"init", projectName}); err == nil {
+			t.Errorf("inferModulePath should error when --module-path is not" +
+				" supplied and GOPATH is empty")
+		}
+	})
+	t.Run("SingleGOPATHEntry", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", testTempDirPrefix)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Error(err)
+			}
+		}()
+		ctx := context.Background()
+		pctx := &processContext{
+			workdir: dir,
+			env:     []string{"GOPATH=" + dir},
+			stdin:   strings.NewReader(""),
+			stdout:  ioutil.Discard,
+			stderr:  ioutil.Discard,
+		}
+		fmt.Println(pctx.env)
+		const projectName = "myspecialproject"
+		if err := run(ctx, pctx, []string{"init", projectName}); err != nil {
```

Comment:
IIUC, this creates a project at `$GOPATH/myspecialproject`, but I believe it should be trying to create the project at `$GOPATH/src/myspecialproject`.

Reply:
Sorry, not sure what you're suggesting here. Why "should" it be adding `src` in the middle? Is this a problem with the test or `init`?

Reply:
I am realizing this is a bug in `gocdk init`. As per [GOPATH](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) documentation, the `src` subdirectory is where source code lives. You can see the difference between `go mod init` and `gocdk init` like this:

```
mkdir $GOPATH/src/xyzzy
cd $GOPATH/src/xyzzy
go mod init
grep '^module' go.mod
```

yields `module xyzzy`, whereas:

```
gocdk init $GOPATH/src/xyzzy
grep '^module' $GOPATH/src/xyzzy/go.mod
```

yields `src/xyzzy`. I can reproduce this at master, so it's not a regression in this PR.

Reply:
This seems like it warrants its own issue, which I can make, but so I can be a little clearer on it: Inferring the module path ought to rely on the given project dir being not just relative to `$GOPATH` but relative to `$GOPATH/src`? (As opposed to current behavior, which is that it only requires it be relative to `$GOPATH`)

Reply:
Opened #2307 to track.